bin/tkldev-changelog

```diff
@@ -80,18 +80,54 @@ which vim > /dev/null || fatal "vim not found; call yourself a developer?!?"
 
 APP=$(basename $PWD)
 COMMON_CHANGES="${FAB_PATH}/common/changes"
-APP_CHANGELOGS="$(tkldev-makefile-info "$PWD/Makefile"  -i | grep '/turnkey/fab/common' | rev | cut -d '/' -f 1 | rev | sed -E "s|^(.*).mk$|/${FAB_PATH}/common/changes/\1.changelog|")"
+APP_CHANGELOGS=$(tkldev-makefile-info "$PWD/Makefile" -i \
+    | grep -E "^${FAB_PATH}/common/" \
+    | sed -En "s|^([a-z0-9/-]+)/mk/turnkey/*([a-z0-9-]+).mk|\1/changes/\2.changelog|p")
 
 if [ -z "$EXTRA_TEXT" ] && [ "$APP" != "core" ]; then
-    EXTRA_TEXT="$(tkldev-changelog-extractor "$PWD/changelog" $APP_CHANGELOGS)"
+    EXTRA_TEXT=$(tkldev-changelog-extractor "$PWD/changelog" "${FAB_PATH}/common/changes/turnkey.changelog" $APP_CHANGELOGS)
 fi
 
 EXTRA_TEXT_FILE="$(mktemp /tmp/tkldev-changelog.XXXXXX)"
 trap "rm -f $EXTRA_TEXT_FILE" EXIT
 echo "$EXTRA_TEXT" > "$EXTRA_TEXT_FILE"
 
-author_name=${FULLNAME:-$DEBFULLNAME}
-author_email=${EMAIL:-$DEBEMAIL}
+author_name=${DEBFULLNAME:-$FULLNAME}
+author_email=${DEBEMAIL:-$EMAIL}
+if [[ -z "$author_name" ]] || [[ -z "$author_email" ]]; then
+    unset changed
+    if [[ -f "$HOME/.bashrc.d/tkldev-author" ]]; then
+        source $HOME/.bashrc.d/tkldev-author
+    fi
+    if [[ -z "$author_name" ]] && [[ -z "$DEBFULLNAME" ]]; then
+        warning "DEBFULLNAME (or FULLNAME) not set in env"
+        read -p "Enter fullname: " author_name
+        changed=true
+    fi
+    if [[ -z "$author_email" ]] && [[ -z "$DEBEMAIL" ]]; then
+        warning "DEBEMAIL (or EMAIL) not set in env"
+        read -p "Enter email: " author_email
+        changed=true
+    fi
+    if [[ "$changed" == 'true' ]]; then
+        read -p "Set permanently? (will save to $HOME/.bashrc.d/tkldev-author) [y]/n: " save
+        case $save in
+            [yY]|yes|YES|"")
+                cat > $HOME/.bashrc.d/tkldev-author <<EOF
+# This file was generated by tkldev-changelog
+export DEBFULLNAME="$author_name"
+export DEBEMAIL="$author_email"
+EOF
+                chmod +x $HOME/.bashrc.d/tkldev-author
+                source $HOME/.bashrc.d/tkldev-author
+                ;;
+            *)
+                export DEBFULLNAME="$author_name"
+                export DEBEMAIL="$author_email"
+                ;;
+        esac
+    fi
+fi
 
 unset set_author set_date new_min_set new_maj_set new edit
 if [[ $# -eq 0 ]]; then
```

Comment:
This looks fine, but why is tkldev-changelog even here again?!

Reply:
I have had a poke around and TBH, I'm still not 100% sure on the rationale?

FWIW it was done by me last year, in this PR: https://github.com/turnkeylinux/tkldev-detective/pull/3 specifically this commit: https://github.com/turnkeylinux/tkldev-detective/commit/80c92e120ba1498329ab3f23cab52bb499b6b5f8

Reply:
Actually, I've just noticed why I included `tkldev-changelog`. It now depends on `tkldev-changelog-extractor` which I also added in the PR I note above. Although that doesn't explain why I added `tkldev-changelog-extractor`? `tkldev-detective` doesn't seem to use either of them, so it seems like a weird thing to do.

I can only assume that I wanted it packaged, but instead of creating a new package with those bits included, I just added it here!? Unless there is any good reason, I should probably create a new package for them and removed it from this package and re/build both.

Reply:
To be honest I do see value in having `tkldev-changelog-extractor` included, changelogs are currently located and tagged by `tkldev-detective`, so even though it doesn't currently use it, perhaps it should be linting changelogs for issues?

Reply:
Ok cool. I'll leave it there then :grin: